---
title: "Gene Set Enrichment Analysis"
author: "Ridvan Cetin"
date: "`r Sys.Date()`"
output: html_notebook
---

We will perform GSEA for one condition.

```{r}
suppressPackageStartupMessages({
  library(dplyr)
  library(fgsea)
  library(ggplot2)
})
```

# Loading the data

## Loading the Differential Gene Expression (DGE) Data from DESeq2

Import DGE results from DESeq2 analysis. Ensure data integrity and format.

```{r}
dge_results <- read.delim(
  "/DESeq2_results_Atf3_cl_1/Late-Mesoderm_resOrdered_enhanced.txt")
# Check for missing values
summary(dge_results)
# Visualize the distribution of gene expression values
hist(dge_results$stat, breaks = 50, main = "Distribution of Gene Expression Values", xlab = "Stat Value")
```

```{r}
ggplot(dge_results, aes(x=stat, y=padj))+geom_point()
```
## Loading the MSigDB (Molecular Signatures Database)

Load gene sets and pathways from MSigDB for enrichment analysis.

```{r}
# MH: hallmark gene sets. Hallmark gene sets summarize and represent specific well-defined biological states or processes and display coherent expression. These gene sets were generated by a computational methodology based on identifying overlaps between gene sets in other MSigDB collections and retaining genes that display coordinate expression.
mh_hallmark <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/mh.all.v2023.2.Mm.symbols.gmt")

# M1: positional gene sets. Gene sets corresponding to mouse chromosome cytogenetic bands.
m1_positional <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m1.all.v2023.2.Mm.symbols.gmt")

# M2: curated gene sets. Gene sets in this collection are curated from various sources, including online pathway databases and the biomedical literature. Many sets are also contributed by individual domain experts. The gene set page for each gene set lists its source. The M2 collection is divided into the following two subcollections: Chemical and genetic perturbations (CGP) and Canonical pathways (CP). 

## CGP: chemical and genetic perturbations. Gene sets represent expression signatures of genetic and chemical perturbations. A number of these gene sets come in pairs: xxx_UP (and xxx_DN) gene set representing genes induced (and repressed) by the perturbation.
m2_curated_cgp <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m2.cgp.v2023.2.Mm.symbols.gmt")

## CP: Canonical pathways. Gene sets from pathway databases. Usually, these gene sets are canonical representations of a biological process compiled by domain experts.

### Biocarta. Canonical Pathways gene sets derived from the BioCarta pathway database.
m2_cp_biocarta <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m2.cp.biocarta.v2023.2.Mm.symbols.gmt")
### Reactome. Canonical Pathways gene sets derived from the Reactome pathway database.
m2_cp_reactome <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m2.cp.reactome.v2023.2.Mm.symbols.gmt")
### WikiPathways. Canonical Pathways gene sets derived from the WikiPathways pathway database.
m2_cp_wikipathways <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m2.cp.wikipathways.v2023.2.Mm.symbols.gmt")

# M3: regulatory target gene sets. Gene sets representing potential targets of regulation by transcription factors or microRNAs. The sets consist of genes grouped by elements they share in their non-protein coding regions. The elements represent known or likely cis-regulatory elements in promoters and 3'-UTRs. The M3 collection is divided into two subcollections: microRNA targets (MIR) and transcription factor targets (TFT).

## miRDB gene sets. Gene sets containing high-confidence gene-level predictions of mouse miRNA targets as catalogued by miRDB v6.0 algorithm
m3_miRDB <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m3.mirdb.v2023.2.Mm.symbols.gmt")
## GTRD gene sets.Genes that share GTRD (Kolmykov et al. 2021) predicted transcription factor binding sites in the region -1000,+100 bp around the TSS for the indicated transcription factor.
m3_gtrd <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m3.gtrd.v2023.2.Mm.symbols.gmt")

# M5: ontology gene sets. Gene sets that contain genes annotated by the same ontology term. The M5 collection is divided into three subcollections derived from the Gene Ontology resource (GO) which contains BP, CC, and MF components.

## GO: Gene Ontology gene sets. All gene sets derived from Gene Ontology
### BP: subset of GO. Gene sets derived from the GO Biological Process ontology.
m5_go_bp <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m5.go.bp.v2023.2.Mm.symbols.gmt")

### CC: subset of GO.	Gene sets derived from the GO Cellular Component ontology.
m5_go_cc <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m5.go.cc.v2023.2.Mm.symbols.gmt")

### MF: subset of GO.	Gene sets derived from the GO Molecular Function ontology
m5_go_mf <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m5.go.mf.v2023.2.Mm.symbols.gmt")

## MPT: tumor phenotype ontology. A subset of ontology terms from the Mammalian Phenotype Ontology database related to cancer specific phenotype terms. details
m5_go_mpt <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m5.mpt.v2023.2.Mm.symbols.gmt")

# M8:  cell type signature gene sets. Gene sets that contain curated cluster markers for cell types identified in single-cell sequencing studies of mouse tissue.
m8_celltype <- fgsea::gmtPathways(
  "~/cmo_chapter_v1/Analysis/4a_gsea/gmt/m8.all.v2023.2.Mm.symbols.gmt")
```


## Creating Ranked Gene List with "stat"

Rank genes based on the "stat" column from the DESeq2 results. Prepare the ranked gene list for GSEA

```{r}
# Removing low expressed genes
dge_results_1 <- dge_results%>%
  filter(baseMean > 15)

# Creating ranked gene file
dge_results_rank <- dge_results_1[c("gene","stat")]

# Addin NA to empty cells
dge_results_rank[dge_results_rank==""] <- NA

# removing NA
dge_results_rank<- na.omit(dge_results_rank)

# ordering the list based on "stat" column
dge_results_rank <- dge_results_rank[order(-dge_results_rank$stat),]

# create "res" object for fgsea
res <- dge_results_rank$stat
names(res) <- dge_results_rank$gene
length(res)
```

# Gene Set Enrichment Analysis (GSEA)

## Running fgsea

Perform GSEA using the fgsea package with the ranked gene list and MSigDB pathways.

```{r,results='hide'}
fgseaRes_mh_hallmark <- 
  fgsea(pathways = mh_hallmark, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m1_positional <-
  fgsea(pathways = m1_positional, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m2_cgp <-
  fgsea(pathways = m2_curated_cgp, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m2_cp_biocarta <-
  fgsea(pathways = m2_cp_biocarta, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m2_cp_reactome <-
  fgsea(pathways = m2_cp_reactome, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m2_cp_wikipathways <-
  fgsea(pathways = m2_cp_wikipathways, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m3_miRDB <-
  fgsea(pathways = m3_miRDB, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m3_gtrd <-
  fgsea(pathways = m3_gtrd, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m5_go_bp <-
  fgsea(pathways = m5_go_bp, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m5_go_cc <-
  fgsea(pathways = m5_go_cc, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m5_go_mf <-
  fgsea(pathways = m5_go_mf, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m5_mpt <-
  fgsea(pathways = m5_go_mpt, stats = res, minSize=15, maxSize=500, nproc = 4)

fgseaRes_m8_celltype <-
  fgsea(pathways = m8_celltype, stats = res, minSize=15, maxSize=500, nproc = 4)


fgseaRes_mh_hallmark <- fgseaRes_mh_hallmark[order(padj),]
fgseaRes_m1_positional <- fgseaRes_m1_positional[order(padj),]
fgseaRes_m2_cgp <- fgseaRes_m2_cgp[order(padj),]
fgseaRes_m2_cp_biocarta <- fgseaRes_m2_cp_biocarta[order(padj),]
fgseaRes_m2_cp_reactome <- fgseaRes_m2_cp_reactome[order(padj),]
fgseaRes_m2_cp_wikipathways <- fgseaRes_m2_cp_wikipathways[order(padj),]
fgseaRes_m3_miRDB <- fgseaRes_m3_miRDB[order(padj),]
fgseaRes_m3_gtrd <- fgseaRes_m3_gtrd[order(padj),]
fgseaRes_m5_go_bp <- fgseaRes_m5_go_bp[order(padj),]
fgseaRes_m5_go_cc <- fgseaRes_m5_go_cc[order(padj),]
fgseaRes_m5_go_mf <- fgseaRes_m5_go_mf[order(padj),]
fgseaRes_m5_mpt <- fgseaRes_m5_mpt[order(padj),]
fgseaRes_m8_celltype <- fgseaRes_m8_celltype[order(padj),]

```

```{r}
saveRDS(fgseaRes_mh_hallmark,"./fgseaRes_mh_hallmark.RDS")
saveRDS(fgseaRes_m1_positional,"./fgseaRes_m1_positional.RDS")
saveRDS(fgseaRes_m2_cgp,"./fgseaRes_m2_cgp.RDS")
saveRDS(fgseaRes_m2_cp_biocarta,"./fgseaRes_m2_cp_biocarta.RDS")
saveRDS(fgseaRes_m2_cp_reactome,"./fgseaRes_m2_cp_reactome.RDS")
saveRDS(fgseaRes_m2_cp_wikipathways,"./fgseaRes_m2_cp_wikipathways.RDS")
saveRDS(fgseaRes_m3_miRDB,"./fgseaRes_m3_miRDB.RDS")
saveRDS(fgseaRes_m3_gtrd,"./fgseaRes_m3_gtrd.RDS")
saveRDS(fgseaRes_m5_go_bp,"./fgseaRes_m5_go_bp.RDS")
saveRDS(fgseaRes_m5_go_cc,"./fgseaRes_m5_go_cc.RDS")
saveRDS(fgseaRes_m5_go_mf,"./fgseaRes_m5_go_mf.RDS")
saveRDS(fgseaRes_m5_mpt,"./fgseaRes_m5_mpt.RDS")
saveRDS(fgseaRes_m8_celltype,"./fgseaRes_m8_celltype.RDS")
```


## Filtering the GSEA Results with padj

Apply a cutoff to filter pathways based on adjusted p-value (padj).

```{r}
fgseaRes_mh_hallmark <-
  fgseaRes_mh_hallmark%>%
  filter(padj<0.05)

fgseaRes_m1_positional <-
  fgseaRes_m1_positional%>%
  filter(padj<0.05)
  
fgseaRes_m2_cgp <-
  fgseaRes_m2_cgp%>%
  filter(padj<0.05)
  
fgseaRes_m2_cp_biocarta <-
  fgseaRes_m2_cp_biocarta%>%
  filter(padj<0.05)

fgseaRes_m2_cp_reactome <-
  fgseaRes_m2_cp_reactome%>%
  filter(padj<0.05)
  
fgseaRes_m2_cp_wikipathways <-
  fgseaRes_m2_cp_wikipathways%>%
  filter(padj<0.05)
  
fgseaRes_m3_miRDB <-
  fgseaRes_m3_miRDB%>%
  filter(padj<0.05)

fgseaRes_m3_gtrd <-
  fgseaRes_m3_gtrd%>%
  filter(padj<0.05)
  
fgseaRes_m5_go_bp <-
  fgseaRes_m5_go_bp%>%
  filter(padj<0.05)
  
fgseaRes_m5_go_cc <-
  fgseaRes_m5_go_cc%>%
  filter(padj<0.05)
  
fgseaRes_m5_go_mf <-
  fgseaRes_m5_go_mf%>%
  filter(padj<0.05)
  
fgseaRes_m5_mpt <-
  fgseaRes_m5_mpt%>%
  filter(padj<0.05)
  
fgseaRes_m8_celltype <-
  fgseaRes_m8_celltype%>%
  filter(padj<0.05)
```

```{r}
saveRDS(fgseaRes_mh_hallmark,"./fgseaRes_mh_hallmark_filtered.RDS")
saveRDS(fgseaRes_m1_positional,"./fgseaRes_m1_positional_filtered.RDS")
saveRDS(fgseaRes_m2_cgp,"./fgseaRes_m2_cgp_filtered.RDS")
saveRDS(fgseaRes_m2_cp_biocarta,"./fgseaRes_m2_cp_biocarta_filtered.RDS")
saveRDS(fgseaRes_m2_cp_reactome,"./fgseaRes_m2_cp_reactome_filtered.RDS")
saveRDS(fgseaRes_m2_cp_wikipathways,"./fgseaRes_m2_cp_wikipathways_filtered.RDS")
saveRDS(fgseaRes_m3_miRDB,"./fgseaRes_m3_miRDB_filtered.RDS")
saveRDS(fgseaRes_m3_gtrd,"./fgseaRes_m3_gtrd_filtered.RDS")
saveRDS(fgseaRes_m5_go_bp,"./fgseaRes_m5_go_bp_filtered.RDS")
saveRDS(fgseaRes_m5_go_cc,"./fgseaRes_m5_go_cc_filtered.RDS")
saveRDS(fgseaRes_m5_go_mf,"./fgseaRes_m5_go_mf_filtered.RDS")
saveRDS(fgseaRes_m5_mpt,"./fgseaRes_m5_mpt_filtered.RDS")
saveRDS(fgseaRes_m8_celltype,"./fgseaRes_m8_celltype_filtered.RDS")
```


## Visualization of GSEA Results

Generate visualizations (e.g., enrichment plots) to summarize GSEA results.

```{r,fig.width=10,fig.height=10}
# Generate the plot
ggplot(fgseaRes_mh_hallmark[1:50,], aes(reorder(pathway, -log10(padj)), -log10(padj), size=size)) +
  geom_point(aes(fill=NES), shape=21) +
  coord_flip() +
  scale_y_continuous(expand = expansion(mult = c(0.05, 0.05))) +
  scale_fill_gradient2(low="red", mid="white", high="blue", midpoint=0) +
  labs(title ="Hallmarks", x = "", y = "-log10(padj)") + 
  theme_minimal() +
  theme(axis.text.x = element_text(size=10),          # Reduce the size of x-axis labels
        axis.text.y.top = element_text(margin = margin(t = 20)),  # Increase gap between x-axis labels and plot
        axis.text.y = element_text(size=10),
        title = element_text(size=18),
        legend.position = "right")
```

```{r,fig.width=15, fig.height=20}
# Function to generate the plot
generate_plot <- function(data, title) {
  ggplot(data[1:100,], aes(reorder(pathway, -log10(padj)), -log10(padj), size=size)) +
    geom_point(aes(fill=NES), shape=21) +
    coord_flip() +
    scale_y_continuous(expand = expansion(mult = c(0.05, 0.05))) +
    scale_fill_gradient2(low="red", mid="white", high="blue", midpoint=0) +
    labs(title = title, x = "", y = "-log10(padj)") + 
    theme_minimal() +
    theme(axis.text.x = element_text(size=10),          # Reduce the size of x-axis labels
          axis.text.x.top = element_text(margin = margin(t = 20)),  # Increase gap between x-axis labels and plot
          axis.text.y = element_text(size=10),
          title = element_text(size=18),
          legend.position = "right")
}

print(generate_plot(fgseaRes_mh_hallmark, "Hallmarks"))
print(generate_plot(fgseaRes_m1_positional, "Positional"))
print(generate_plot(fgseaRes_m2_cgp, "CGP"))
print(generate_plot(fgseaRes_m2_cp_biocarta, "BioCarta"))
print(generate_plot(fgseaRes_m2_cp_reactome, "Reactome"))
print(generate_plot(fgseaRes_m2_cp_wikipathways, "WikiPathways"))
print(generate_plot(fgseaRes_m3_miRDB, "miRDB"))
print(generate_plot(fgseaRes_m3_gtrd, "GTRD"))
print(generate_plot(fgseaRes_m5_go_bp, "GO BP"))
print(generate_plot(fgseaRes_m5_go_cc, "GO CC"))
print(generate_plot(fgseaRes_m5_go_mf, "GO MF"))
print(generate_plot(fgseaRes_m5_mpt, "MPT"))
print(generate_plot(fgseaRes_m8_celltype, "Celltype"))

# Generate and save plots
pdf("gsea_results_plots.pdf",width = 15, height = 20)
print(generate_plot(fgseaRes_mh_hallmark, "Hallmarks"))
print(generate_plot(fgseaRes_m1_positional, "Positional"))
print(generate_plot(fgseaRes_m2_cgp, "CGP"))
print(generate_plot(fgseaRes_m2_cp_biocarta, "BioCarta"))
print(generate_plot(fgseaRes_m2_cp_reactome, "Reactome"))
print(generate_plot(fgseaRes_m2_cp_wikipathways, "WikiPathways"))
print(generate_plot(fgseaRes_m3_miRDB, "miRDB"))
print(generate_plot(fgseaRes_m3_gtrd, "GTRD"))
print(generate_plot(fgseaRes_m5_go_bp, "GO BP"))
print(generate_plot(fgseaRes_m5_go_cc, "GO CC"))
print(generate_plot(fgseaRes_m5_go_mf, "GO MF"))
print(generate_plot(fgseaRes_m5_mpt, "MPT"))
print(generate_plot(fgseaRes_m8_celltype, "Celltype"))
dev.off()
```

```{r}
fgseaRes_mh_hallmark_down <-
  fgseaRes_mh_hallmark%>%
  filter(NES< 0)

fgseaRes_mh_hallmark_down$pathway


fgseaRes_mh_hallmark_up <-
  fgseaRes_mh_hallmark%>%
  filter(NES> 0)

fgseaRes_mh_hallmark_up$pathway
```

```{r}
fgseaRes_m2_cp_reactome_down <-
  fgseaRes_m2_cp_reactome%>%
  filter(NES< 0)

fgseaRes_m2_cp_reactome_down$pathway


fgseaRes_m2_cp_reactome_up <-
  fgseaRes_m2_cp_reactome%>%
  filter(NES> 0)

fgseaRes_m2_cp_reactome_up$pathway
```

```{r}
fgseaRes_m5_go_bp_down <-
  fgseaRes_m5_go_bp%>%
  filter(NES< 0)

fgseaRes_m5_go_bp_down$pathway


fgseaRes_m5_go_bp_up <-
  fgseaRes_m5_go_bp%>%
  filter(NES> 0)

fgseaRes_m5_go_bp_up$pathway
```

## Selecting

```{r}
# Ensure pathways exists in the global environment
if (!exists("pathways")) {
  pathways <- list()
}

generate_enrichment_plot <- function(pathway_name, fgsea_result_table, ranked_list,
                                     mh_hallmark, output_directory = "./", 
                                     plot_width = 5, plot_height = 3) {
  # Extract the specific row for the pathway of interest
  stats <- fgsea_result_table[fgsea_result_table$pathway == pathway_name, ]
  
  if(nrow(stats) == 0) {
    stop("Pathway not found in the fgsea result table.")
  }

  # Create the enrichment plot
  plot <- plotEnrichment(mh_hallmark[[pathway_name]], ranked_list, ticksSize = 0.2) +
    labs(title = pathway_name,
         caption = paste("padj: ", formatC(stats$padj, format = "e", digits = 2),
                         "NES: ", round(stats$NES, 2),
                         "Size: ", stats$size)) +
    theme(plot.title = element_text(face = "bold", hjust = 0.5),
          axis.title.x = element_text(size = 12),
          axis.title.y = element_text(size = 12))

  # Save the plot to a PDF file
  ggsave(
    filename = paste0(output_directory, "/", pathway_name, ".pdf"),
    plot = plot,
    width = plot_width,
    height = plot_height,
    units = "in",
    dpi = 300,
    limitsize = TRUE
  )
  
  print(plot)
  
  # Update the pathways list in the global environment
  pathways[[pathway_name]] <<- stats
  
  # Return the leading edge
  return(stats$leadingEdge)
}
```

```{r,fig.width=5,fig.height=3}
generate_enrichment_plot("HALLMARK_TGF_BETA_SIGNALING",
                         fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
generate_enrichment_plot("HALLMARK_TNFA_SIGNALING_VIA_NFKB", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
generate_enrichment_plot("HALLMARK_KRAS_SIGNALING_UP",
                         fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
generate_enrichment_plot("HALLMARK_WNT_BETA_CATENIN_SIGNALING",
                         fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_NOTCH_SIGNALING", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_MITOTIC_SPINDLE", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_APOPTOSIS", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_ESTROGEN_RESPONSE_EARLY", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_HYPOXIA", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_INTERFERON_GAMMA_RESPONSE", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_INTERFERON_ALPHA_RESPONSE", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_IL6_JAK_STAT3_SIGNALING", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_OXIDATIVE_PHOSPHORYLATION", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_GLYCOLYSIS", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_FATTY_ACID_METABOLISM", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_ADIPOGENESIS", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_BILE_ACID_METABOLISM", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_PEROXISOME", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_PROTEIN_SECRETION", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_MTORC1_SIGNALING", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_UV_RESPONSE_DN", fgseaRes_mh_hallmark, res, mh_hallmark)
```

HALLMARK_ANGIOGENESIS HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION HALLMARK_MYOGENESIS

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_ANGIOGENESIS", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_EPITHELIAL_MESENCHYMAL_TRANSITION", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_MYOGENESIS", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_UV_RESPONSE_DN", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_UV_RESPONSE_DN", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_UV_RESPONSE_DN", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_UV_RESPONSE_DN", fgseaRes_mh_hallmark, res, mh_hallmark)
```

```{r,fig.width=5,fig.height=3}
# Example usage
generate_enrichment_plot("HALLMARK_UV_RESPONSE_DN", fgseaRes_mh_hallmark, res, mh_hallmark)
```


```{r}
library(stringr)
```

```{r}
pathways <- readRDS("./pathways.RDS")
```

```{r}
fgseaRes_mh_hallmark_filtered <- readRDS("./fgseaRes_mh_hallmark_filtered.RDS")
fgseaRes_m5_go_bp_filtered <- readRDS("./fgseaRes_m5_go_bp_filtered.RDS")
fgseaRes_m2_cp_reactome <- readRDS("./fgseaRes_m2_cp_reactome_filtered.RDS")
```
```{r}
gsea <- rbind(fgseaRes_mh_hallmark_filtered,fgseaRes_m5_go_bp_filtered,fgseaRes_m2_cp_reactome)
```



```{r,fig.width=6.5,fig.height=5.5}
# Create a sample dataset for reproducibility
set.seed(123)
# Define the wrap_text function with underscore replacement
wrap_text <- function(y) str_wrap(gsub("_", " ", y), width = 45)

# Create the plot
pl_emp_hall <-
  ggplot(fgseaRes_mh_hallmark_filtered[1:20,], aes(y=reorder(pathway, -log10(padj)), x=-log10(padj), size=size)) +
  geom_point(aes(fill=NES), shape=21) +
  scale_fill_gradient2(low="red", mid="white", high="blue", midpoint=0) +
  labs(title ="EMP GSEA:HALLMARK",y="") + 
  theme_classic() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 12, color = "black", face = "bold"),
    axis.title.x = element_text(size = 8),
    axis.title.y = element_text(size = 8),
    axis.text = element_text(size = 8),
    axis.text.y = element_text(size = 8, margin = margin(t = 0, r = 0, b = 0, l = 0)),
    legend.position = "inside",
    legend.position.inside = c(0.92, 0.45),
    legend.background = element_rect(fill = "transparent"),
    legend.title = element_text(size = 8),
    legend.text = element_text(size = 8),
    axis.line = element_line(size = 0.2)
  ) +
  scale_y_discrete(labels = function(y) wrap_text(y)) +
  labs(size="Genes")

# Display the plot
print(pl_emp_hall)
```


```{r,fig.width=6.5,fig.height=5.5}
# Create a sample dataset for reproducibility
set.seed(123)
# Define the wrap_text function with underscore replacement
wrap_text <- function(y) str_wrap(gsub("_", " ", y), width = 45)

# Create the plot
pl_emp_hall <-
  ggplot(fgseaRes_mh_hallmark_filtered[1:20,], aes(y=reorder(pathway, -log10(padj)), x=-log10(padj), size=size)) +
  geom_point(aes(fill=NES), shape=21) +
  scale_fill_gradient2(low="red", mid="white", high="blue", midpoint=0) +
  labs(title ="EMP GSEA:HALLMARK",y="") + 
  theme_classic() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 12, color = "black", face = "bold"),
    axis.title.x = element_text(size = 8),
    axis.title.y = element_text(size = 8),
    axis.text = element_text(size = 8),
    axis.text.y = element_text(size = 8, margin = margin(t = 0, r = 0, b = 0, l = 0)),
    legend.position = "inside",
    legend.position.inside = c(0.92, 0.45),
    legend.background = element_rect(fill = "transparent"),
    legend.title = element_text(size = 8),
    legend.text = element_text(size = 8),
    axis.line = element_line(size = 0.2)
  ) +
  scale_y_discrete(labels = function(y) wrap_text(y)) +
  labs(size="Genes")

# Display the plot
print(pl_emp_hall)
```

```{r,fig.width=6.5,fig.height=5.5}
# Create a sample dataset for reproducibility
set.seed(123)
# Define the wrap_text function with underscore replacement
wrap_text <- function(y) str_wrap(gsub("_", " ", y), width = 45)

# Create the plot
pl_emp_gobp <-
  ggplot(fgseaRes_m5_go_bp_filtered[1:20,], aes(y=reorder(pathway, -log10(padj)), x=-log10(padj), size=size)) +
  geom_point(aes(fill=NES), shape=21) +
  scale_fill_gradient2(low="red", mid="white", high="blue", midpoint=0) +
  labs(title ="EMP GSEA:GO-BP",y="") + 
  theme_classic() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 12, color = "black", face = "bold"),
    axis.title.x = element_text(size = 8),
    axis.title.y = element_text(size = 8),
    axis.text = element_text(size = 8),
    axis.text.y = element_text(size = 8, margin = margin(t = 0, r = 0, b = 0, l = 0)),
    legend.position = "inside",
    legend.position.inside = c(0.92, 0.45),
    legend.background = element_rect(fill = "transparent"),
    legend.title = element_text(size = 8),
    legend.text = element_text(size = 8),
    axis.line = element_line(size = 0.2)
  ) +
  scale_y_discrete(labels = function(y) wrap_text(y)) +
  labs(size="Genes")

# Display the plot
print(pl_emp_gobp)
```

```{r,fig.width=6.5,fig.height=5.5}
# Create a sample dataset for reproducibility
set.seed(123)
# Define the wrap_text function with underscore replacement
wrap_text <- function(y) str_wrap(gsub("_", " ", y), width = 45)

# Create the plot
pl_emp_reactome <- 
  ggplot(fgseaRes_m2_cp_reactome[1:20,], aes(y=reorder(pathway, -log10(padj)), x=-log10(padj), size=size)) +
  geom_point(aes(fill=NES), shape=21) +
  scale_fill_gradient2(low="red", mid="white", high="blue", midpoint=0) +
  labs(title ="EMP GSEA-Reactome",y="") + 
  theme_classic() +
  theme(
    plot.title = element_text(hjust = 0.5, size = 12, color = "black", face = "bold"),
    axis.title.x = element_text(size = 8),
    axis.title.y = element_text(size = 8),
    axis.text = element_text(size = 8),
    axis.text.y = element_text(size = 8, margin = margin(t = 0, r = 0, b = 0, l = 0)),
    legend.position = "inside",
    legend.position.inside = c(0.92, 0.45),
    legend.background = element_rect(fill = "transparent"),
    legend.title = element_text(size = 8),
    legend.text = element_text(size = 8),
    axis.line = element_line(size = 0.2)
  ) +
  scale_y_discrete(labels = function(y) wrap_text(y)) +
  labs(size="Genes")

# Display the plot
print(pl_emp_reactome)

```

```{r}
sessionInfo()
```
















